sktime/forecasting/mapa.py

```diff
@@ -0,0 +1,770 @@
+"""MAPA (Multiple Aggregation Prediction Algorithm) Forecaster."""
+
+import multiprocessing
+from concurrent.futures import ProcessPoolExecutor
+from copy import deepcopy
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.ets import AutoETS
+from sktime.forecasting.exp_smoothing import ExponentialSmoothing
+from sktime.utils.datetime import _get_freq
+from sktime.utils.validation.series import check_series
+
+
+class MAPAForecaster(BaseForecaster):
+    """MAPA (Multiple Aggregation Prediction Algorithm) Forecaster."""
+
+    def __init__(
+        self,
+        comb: str = "w.mean",
+        hybrid: bool = True,
+        conf_lvl: Optional[float] = None,
+        sp: Optional[int] = None,
+        etstype: str = "ets",  # Add this line
+    ):
+        super().__init__()
+        self.comb = comb
+        self.hybrid = hybrid
+        self.conf_lvl = conf_lvl
+        self.sp = sp
+        self.etstype = etstype  # Add this line
+        self._y = None
+        self._mapafit = None
+        self._is_fitted = False
+
+    @property
+    def y_(self):
+        """Returns _y."""
+        return self._y
+
+    @property
+    def mapafit_(self):
+        """Returns _mapafit."""
+        return self._mapafit
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data."""
+        self._y = check_series(y).copy()
```

Comment:
you do not need to `check_series` or `check_is_fitted`, this is also done by the boilerplate - have a look at the `forecasting_supersimple` extension template